pinot-spi/src/main/java/org/apache/pinot/spi/config/table/QueryConfig.java

```diff
@@ -56,13 +56,27 @@ public class QueryConfig extends BaseJsonConfig {
   // Indicates the maximum length of the serialized response per server for a query.
   private final Long _maxServerResponseSizeBytes;
 
-  @JsonCreator
+  // Table config override for query cache
+  private final CacheConfig _cacheConfig;
+
   public QueryConfig(@JsonProperty("timeoutMs") @Nullable Long timeoutMs,
       @JsonProperty("disableGroovy") @Nullable Boolean disableGroovy,
       @JsonProperty("useApproximateFunction") @Nullable Boolean useApproximateFunction,
       @JsonProperty("expressionOverrideMap") @Nullable Map<String, String> expressionOverrideMap,
       @JsonProperty("maxQueryResponseSizeBytes") @Nullable Long maxQueryResponseSizeBytes,
       @JsonProperty("maxServerResponseSizeBytes") @Nullable Long maxServerResponseSizeBytes) {
+    this(timeoutMs, disableGroovy, useApproximateFunction, expressionOverrideMap, maxQueryResponseSizeBytes,
+        maxServerResponseSizeBytes, null);
+  }
+
+  @JsonCreator
+  public QueryConfig(@JsonProperty("timeoutMs") @Nullable Long timeoutMs,
```

Comment:
(minor) You only need to annotate one constructor. If you decide to keep the other constructor, then no need to change the usage in other classes